client/src/components/create-recipe/CreateForm.test.js

```diff
@@ -0,0 +1,183 @@
+        const { getByTestId, getByText } = render(<CreateForm />, {wrapper: AuthProvider});
+
+        await act(() => Promise.resolve());
+        
+        const file = new File(['foo'], 'foo.txt', {type: 'text/plain'})
+
+        const imageInput = getByTestId('image-input');
+        fireEvent.change(imageInput, {target: {files: [file]}});
+
+        await act(() => Promise.resolve());
+
+        expect(getByText('unable to convert image file')).toBeInTheDocument();
+    })
+    test('uploading a image file should display an image onto the Dom', async () => {
+        const file = new File(['(⌐□_□)'], 'coolGuy.png', {type: 'image/png'});
+        const fakeBase64 = 'data:image/jpeg;base64,/9j/4AAQSkZJRgABAQAAAQABAAD/2wBDAAYEBQUFBAYFB'
+        ImageCompressor.mockResolvedValue(file);
+        ToBase64.mockResolvedValue(fakeBase64);
+
+        const { getByTestId } = render(<CreateForm />, {wrapper: AuthProvider});
+
+        await act(() => Promise.resolve());
+
+        fireEvent.change(getByTestId('image-input'), {target: {files: [file]}});
+
+        await act(() => Promise.resolve());
+
+        expect(getByTestId('displayed-image')).toBeInTheDocument();
+    })
+})
+
+describe('Create recipe button', () => {
+
+    beforeEach(() => {
+        const authenticatedResponse = {isAuthenticated: true, user: {username: 'Bpun1p'}};
+        AuthService.isAuthenticated.mockResolvedValue(authenticatedResponse);
+    });
+
+    test('when click with no inputs will display a error message in the DOM', async () => {
+        const { getByText } = render(<CreateForm />, {wrapper: AuthProvider});
+
+        await act(() => Promise.resolve());
+
+        fireEvent.click(getByText('Create'));
+
+        expect(getByText('Please fill in all fields')).toBeInTheDocument();
+    });
+    test('when clicked with some filled inputs will display an error message in the DOM', async () => {
+        const { getByText, getByTestId, getAllByTestId } = render(<CreateForm />, {wrapper: AuthProvider});
+
+        await act(() => Promise.resolve());
+        const ingredientField = getAllByTestId('content-text-field')[0];
+        const ingredientButton = getAllByTestId('add-content-button')[0];
+        fireEvent.change(getByTestId('recipe-name-text-field'), {target: {value: 'Chicken Pot Pie'}});
+        fireEvent.change(getByTestId('description-text-field'), {target: {value: 'A delicious chicken pie made from scratch with carrots, peas and celery.'}});
+        fireEvent.change(ingredientField, {target: {value: 'chicken breast'}});
+        fireEvent.click(ingredientButton);
+
+        fireEvent.click(getByText('Create'));
+
+        expect(getByText('Please fill in all fields')).toBeInTheDocument();
+    });
+    test('when clicked with fully filled input will send user back to their dashboard', async () => {
+        const file = new File(['(⌐□_□)'], 'coolGuy.png', {type: 'image/png'});
```

Comment:
png naming should be 'cool-guy'? if following previous naming scheme

Reply:
adjustments made